### PR TITLE
tagged tangential discontinuity now tests expected TD position

### PR DESCRIPTION
**Pull request by nicolasaunai**
update the tdtagged1d functional test.
The test now produces the figure of the paper 

image

and also checks wether the TD position at that time is what is expected
Data around the rightward TD is extracted and used to fit an tanh((x-x0)/L) where x0 and L are expected to be around 172 and 1.

Small test here gives (data blue, fit orange) : 

image

with L=1.18 and x0=171.7, not top bad.

### CODE REVIEW

File: tests/functional/tdtagged/td1dtagged.py [hunk 1]
```diff
@@ -112,8 +218,10 @@ def vthz(x):
 
 
 
-    timestamps = all_timestamps(gv.sim)
+    sim = ph.global_vars.sim
 
+    timestamps = np.arange(0, sim.final_time, sim.time_step)
```
Comment (PhilipDeegan) [inline tests/functional/tdtagged/td1dtagged.py:223]: intentional/preferred/copy paste?

File: tests/functional/tdtagged/td1dtagged.py [hunk 2]
```diff
@@ -132,12 +240,114 @@ def vthz(x):
...
+        raise RuntimeError(f"x0 (={x0}) too far from 172")
+
+
+
+
+
 
 def main():
-    config()
+    from pybindlibs import mpi_rank
```
Comment (PhilipDeegan) [inline tests/functional/tdtagged/td1dtagged.py:336]: this works?

Comment (nicolasaunai) [reply tests/functional/tdtagged/td1dtagged.py:336]: why not?

File: tests/functional/tdtagged/td1dtagged.py (hunk 1 above)
Comment (nicolasaunai) [reply tests/functional/tdtagged/td1dtagged.py:223]: need redo..

File: src/amr/messengers/hybrid_hybrid_messenger_strategy.h [hunk 3]
```diff
@@ -8,6 +8,7 @@
 #include "amr/data/field/time_interpolate/field_linear_time_interpolate.h"
 #include "amr/data/particles/refine/particles_data_split.h"
 #include "amr/data/particles/refine/split.h"
+#include "amr/messengers/messenger_info.h"
```
[the remark this answers is not in the document]
Comment (nicolasaunai) [reply src/amr/messengers/hybrid_hybrid_messenger_strategy.h:11]: vim complained!

File: tests/functional/tdtagged/td1dtagged.py (hunk 2 above)
Comment (PhilipDeegan) [reply tests/functional/tdtagged/td1dtagged.py:336]: image

Comment (nicolasaunai) [reply tests/functional/tdtagged/td1dtagged.py:336]: ok
```python
from pybindlibs.cpp import mpi_rank
```
should do it

File: tests/functional/alfven_wave/alfven_wave1d.py [hunk 4]
```diff
@@ -156,6 +156,8 @@ def phase_speed(run_path, ampl, xmax):
 
 def main():
     from pybindlibs.cpp import mpi_rank
+    from pyphare.pharesee.run import Run
```
Comment (PhilipDeegan) [inline tests/functional/alfven_wave/alfven_wave1d.py:159]: redundant since #533 maybe

File: tests/functional/tdtagged/td1dtagged.py [hunk 5]
```diff
@@ -111,9 +152,11 @@ def vthz(x):
     ElectronModel(closure="isothermal", Te=0.12)
 
 
+    sim = ph.global_vars.sim
 
-    timestamps = all_timestamps(gv.sim)
-
+    dtdiag = sim.time_step
+    ndiag  = sim.final_time/dtdiag+1
+    timestamps = dtdiag * np.arange(ndiag)
```
Comment (PhilipDeegan) [inline tests/functional/tdtagged/td1dtagged.py:159]: is this better?

File: tests/functional/alfven_wave/alfven_wave1d.py (hunk 4 above)
Comment (nicolasaunai) [reply tests/functional/alfven_wave/alfven_wave1d.py:159]: is removed in #535

File: tests/functional/tdtagged/td1dtagged.py (hunk 5 above)
Comment (nicolasaunai) [reply tests/functional/tdtagged/td1dtagged.py:159]: than?

Comment (PhilipDeegan) [reply tests/functional/tdtagged/td1dtagged.py:159]: "timestamps = all_timestamps(gv.sim)"

Comment (nicolasaunai) [reply tests/functional/tdtagged/td1dtagged.py:159]: right! forgot about it probably. will change

**Review by PhilipDeegan — APPROVED** [review]

**Pull request merged**